docs/rfc/009-bit-byte-conversions.md

```diff
@@ -8,237 +8,330 @@ The Aleo Team.
+and this causes the conversions to use the same number of bits and bytes as the field conversions,
+namely `N` is 253 and `M` is 32.
+
+A character can be always converted to a (little or big endian) sequence of bits or bytes.
+However, analogously to field and group elements,
+attempting to convert to character a sequence whose numeric value exceeds 10FFFFh
+causes an error in Leo.
+This is the case also if `N` and `M` are reduced to be 21 and 3.
 
-## Transition to Methods
+```ts
+let x: char = 'A';
+console.assert(x.to_bits_le() == [true, false, false, false, false, false, true, false, ...[false; 245]];
+console.assert(x.to_bits_be() == [...[false; 245], false, true, false, false, false, false, false, true];
+console.assert(x.to_bytes_le() == [65, ...[0; 31]];
+console.assert(x.to_bytes_be() == [...[0; 31], 65];
+let y: char = char::from_bits_le([true; 21]); // error
+let y: char = char::from_bytes_le([255, 255, 255]); // error
+```
 
-Once a separate proposal for adding methods to Leo scalar types is realized,
-we may want to turn the global functions listed above into methods,
-deprecating the global functions, and eventually eliminating them.
+### Conversions with Booleans
+
+Converting between booleans and boolean sequences amount to
+converting between the boolean and the singleton array that contains the boolean;
+in this case, there is no difference between little and big endian order.
+These conversions are available for completeness, but may not be very useful in practice.
 
-Conversions to bits or bytes will be instance methods of the integer types,
-e.g. `u8` will include an instance method `to_bits_le` that takes no arguments and that returns a `[bool; 8]`.
-Example:
 ```ts
-let int: u8 = 12;
-let bits: [bool; 8] = int.to_bits_le();
-console.assert(bits == [false, false, true, true, false, false, false, false]); // 00110000 (little endian)
+console.assert(false.to_bits_le() == [false]);
+console.assert(false.to_bits_be() == [false]);
+console.assert(bool::from_bits_le([true]) == true);
+console.assert(bool::from_bits_be([true]) == true);
 ```
 
-Conversions from bits or bytes will be static methods of the integer types,
-e.g. `u8` will include a static metod `from_bits_le` that takes a `[bool; 8]` argument and returns a `u8`.
-Example:
+### Conversions with Addresses
+
+A Leo address is a sequence of 63 lowercase letters and decimal digits that starts with `aleo1`.
+In this form, it is not a number or combination of numbers of any sort.
+
+However, as documented elsewhere,
+an address is essentially a public key that consists of the _x_ coordinate of a curve point.
+The `aleo1...` sequence of 63 characters can be derived from that,
+and the value of the _x_ coordinate can be derived from a (properly formed) `aleo1...` sequence of 63 characters.
+
+Thus, an address can be treated in the same way as a field element for the purpose of bit/byte conversions.
+This is what the bit/byte conversion functions for addresses do in Leo.
```

Comment:
It would be good to clarify what happens here.  The 58-char Bech32 data should become 58x5 = 290 bits, so how does that fit in a field?  I suspect there is some checksum removal going on.

Reply:
Bech32 is actually a decodable 20-byte array. Perhaps, we could figure out how to “decode” these bytes in Leo and use them instead of chars?

Reply:
Yes, I agree that we should flesh out what happens here. I think the information can be gathered from the snarkVM code in the gadget (the conversion is implemented there).

Using the 20 bytes (= 160 bits) would be more efficient that using the x coordinate of the public address (253 bits). Indeed, the address appears as a sequence of 58 lowercase letters and digits.

@damirka Would you be interested in investigating what snarkVM does and whether we could use the 20-byte approach you mention?